java/src/main/java/com/livefyre/protobuf/rpc/AsyncRpc.java

```diff
@@ -0,0 +1,32 @@
+package com.livefyre.protobuf.rpc;
+
+import com.google.protobuf.Message;
+import com.google.protobuf.RpcCallback;
+
+import java.util.concurrent.CompletableFuture;
+
+public class AsyncRpc<T extends Message> {
```

Comment:
moving this into the Controller is problematic b/c of generics, if you have suggestions/ideas feel free to share